PWGHF/D2H/TableProducer/candidateSelectorB0ToDPiReduced.cxx

```diff
@@ -163,15 +156,15 @@ struct HfCandidateSelectorB0ToDPiReduced {
         registry.fill(HIST("hSelections"), 2 + SelectionStep::RecoTopol, ptCandB0);
       }
 
-      // checking if selectionFlagD and usePid are in sync
-      if (!selectionFlagDAndUsePidInSync) {
-        hfSelB0ToDPiCandidate(statusB0ToDPi);
-        continue;
-      }
       // track-level PID selection
-      if (usePid) {
+      if (usePionPid) {
         auto trackPi = hfCandB0.template prong1_as<TracksPion>();
-        int pidTrackPi = selectorPion.statusTpcAndTof(trackPi);
+        int pidTrackPi{TrackSelectorPID::Status::NotApplicable};
```

Comment:
Hi @fgrosa ! Everything looks okay, thanks :)
Maybe we could avoid an `if` condition by doing this
```
int pidTrackPi = selectorPion.statusTpcAndTof(trackPi);
if (usePionPid == 2) {
  pidTrackPi = selectorPion.statusTpcAndTof(trackPi);
}
```
as, if the condition `usePionPid` is fulfilled, it means that it can only be `1` or `2` (the case `=1` would be our default condition).

Reply:
Hi @AlexBigO thanks for the feedback! I still have a preference to keep it as it is (if you prefer I can put `if else` instead of `if else if`) because with your suggestion, if `usePionPid == 2` then we compute the status twice instead of once (it is not a simple return, see https://github.com/AliceO2Group/O2Physics/blob/52f1faf340047b3abbf92f59adc23af04d530298/Common/Core/TrackSelectorPID.h#L390-L442)

Reply:
Indeed, my option is counter-productive :) Yes, then going for `if then` is fine, thanks!

Reply:
Done in https://github.com/AliceO2Group/O2Physics/pull/4471/commits/723376e81657e64c6935dda73e2b6e9c76dce9c0